src/main/scala/Musica.scala

```diff
@@ -1,18 +1,18 @@
 package object Musica {
   trait Nota {
-    val notas = List(C, Cs, D, Ds, E, F, Fs, G, Gs, A, As, B)
+    val notas: List[Nota] = List(C, Cs, D, Ds, E, F, Fs, G, Gs, A, As, B)
 
-    def sostenido: Nota = (notas ++ notas).apply(notas.indexOf(this) + 1)
+    def sostenido: Nota = notas.lift(notas.indexOf(this) + 1).getOrElse(notas.head)
 
-    def bemol: Nota = (notas ++ notas).apply(notas.lastIndexOf(this) - 1)
```

Comment:
Si hubiera querido seguir con esta forma sería `notas.length + notas.lastIndexOf(this) - 1` btw